### DISCUSSION TITLE
What is the "New Metadata System"?

### DISCUSSION THREAD

**Discussion by jpjonte** [api-platform/core#5048] [General]
The upgrade guide mentions the "New Metadata System" and how to enable it.
But there is no explanation anywhere in the docs what the new system actually entails.

I saw that the ResourceMetadataFactoryInterface is now deprecated in favor of ResourceMetadataCollectionFactoryInterface, but there is no upgrade guide to convert existing factories to the new interface.

I'm sure I'll figure out what needs to be done to get my existing code working with API Platform 3, but the docs feel a little thin.

**Comment by soyuka** [comment]
It allows to have multiple `ApiResource` on a class. This was not possible in 2.x and in 3.x allows to improve the usage and developer experience of subresources. Indeed, you need to check our code to see how metadata factories work.
